Neos.Flow/Classes/ObjectManagement/Proxy/ObjectSerializationTrait.php

```diff
@@ -131,7 +131,6 @@ private function Flow_searchForEntitiesAndStoreIdentifierArray($path, $propertyV
                 'identifier' => $identifier,
                 'entityPath' => $path
             ];
-            $this->$originalPropertyName = Arrays::setValueByPath($this->$originalPropertyName, $path, null);
```

Comment:
See https://github.com/neos/flow-development-collection/pull/2012/files#r429575763

Reply:
I _think_ the unset was in place to make sure persisting worked. The actual entities need to be encoded, but keeping them means they _get serialized as well_, doubling data and potentially having bad side-effects. I can imagine this even dates back to our own persistence layer before Doctrine 2… So, it might not be needed anymore.

That being said, the whole concept of `Flow_searchForEntitiesAndStoreIdentifierArray` might be useless by now?!

Reply:
Not entirely, no. At least if you use Flowpack.JobQueue and the @Defer - Annotation, you will see, that just passing a Doctrine collection will lead to detached entities in the Job, while a DTO with a collection property will not. So it is still "of general use"